Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /staff/notifications
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 132, in dispatch
    user_test_result = self.get_test_func()()
                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 119, in test_func
    raise NotImplementedError(
NotImplementedError: NotificationView is missing the implementation of the test_func() method.
"GET /staff/notifications HTTP/1.1" 500 145
Internal Server Error: /
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Home/views.py", line 18, in get
    return render(request,'home.html',context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/loader_tags.py", line 210, in render
    return template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 173, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 113, in url
    return self.handle_simple(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 129, in handle_simple
    return staticfiles_storage.url(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 203, in url
    return self._url(self.stored_name, name, force)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 182, in _url
    hashed_name = hashed_name_func(*args)
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 516, in stored_name
    raise ValueError(
ValueError: Missing staticfiles manifest entry for 'img/black.png'
"GET / HTTP/1.1" 500 145
Internal Server Error: /
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Home/views.py", line 18, in get
    return render(request,'home.html',context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/loader_tags.py", line 210, in render
    return template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 173, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 113, in url
    return self.handle_simple(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/templatetags/static.py", line 129, in handle_simple
    return staticfiles_storage.url(path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 203, in url
    return self._url(self.stored_name, name, force)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 182, in _url
    hashed_name = hashed_name_func(*args)
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 516, in stored_name
    raise ValueError(
ValueError: Missing staticfiles manifest entry for 'img/black.png'
"GET / HTTP/1.1" 500 145
Internal Server Error: /staff/music_sheets
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(100)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Staff/views.py", line 134, in post
    Copies.objects.create(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.DataError: value too long for type character varying(100)

"POST /staff/music_sheets HTTP/1.1" 500 173013
Internal Server Error: /staff/music_sheets
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(100)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Staff/views.py", line 145, in post
    serializer.save()
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 962, in create
    instance = ModelClass._default_manager.create(**validated_data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.DataError: value too long for type character varying(100)

"POST /staff/music_sheets HTTP/1.1" 500 185913
Internal Server Error: /staff/music_sheets
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(100)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Staff/views.py", line 136, in post
    serializer.save()
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 962, in create
    instance = ModelClass._default_manager.create(**validated_data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.DataError: value too long for type character varying(100)

"POST /staff/music_sheets HTTP/1.1" 500 185765
Internal Server Error: /staff/music_sheets
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(100)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Staff/views.py", line 136, in post
    serializer.save()
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 962, in create
    instance = ModelClass._default_manager.create(**validated_data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.DataError: value too long for type character varying(100)

"POST /staff/music_sheets HTTP/1.1" 500 185765
Internal Server Error: /staff/music_sheets
Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(100)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/CatholicSong-main/Staff/views.py", line 137, in post
    serializer.save()
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 212, in save
    self.instance = self.create(validated_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/rest_framework/serializers.py", line 962, in create
    instance = ModelClass._default_manager.create(**validated_data)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/ngoga/Desktop/Web/Apis/envs/catholicSongenv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.DataError: value too long for type character varying(100)

"POST /staff/music_sheets HTTP/1.1" 500 185772
